### PR TITLE
modules/SceLibKernel: sceKernelGetThreadExitStatus implemented

### PR DESCRIPTION
**Pull request by bookmist**
Code from too old and difficult to merge pull request #796

### CODE REVIEW

**Comment by Zangetsu38** [comment]
If you can also look for re implément your code for  sceKernelClearEventFlag is also very request

**Comment by bookmist** [comment]
> If you can also look for re implément your code for sceKernelClearEventFlag is also very request

sceKernelClearEventFlag is already implemented. It's why I made new pull request.

**Comment by Zangetsu38** [comment]
> 
> 
> > If you can also look for re implément your code for sceKernelClearEventFlag is also very request
> 
> sceKernelClearEventFlag is already implemented. It's why I made new pull request.

Oups yes sorry, i have forget lol, is caused by missing "bool do_clear", is not needed ? 
But sceKernelGetLwMutexInfoById and sceKernelGetLwMutexInfo keep missing

**Comment by Zangetsu38** [comment]
Also, if yopu have any idea for how implement "sceKernelCheckCallback" very request here :/

File: vita3k/modules/SceLibKernel/SceLibKernel.cpp [hunk 1]
```diff
@@ -1101,8 +1101,15 @@ EXPORT(int, sceKernelGetThreadEventInfo) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceKernelGetThreadExitStatus) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceKernelGetThreadExitStatus, SceUID thid) {
+    const ThreadStatePtr thread = lock_and_find(thid ? thid : thread_id, host.kernel.threads, host.kernel.mutex);
```
Comment (scribam) [inline vita3k/modules/SceLibKernel/SceLibKernel.cpp:1105]: Does this condition `thid ? thid : thread_id` really exists?
I was expecting the following to be true : `sceKernelGetThreadExitStatus(NULL) == SCE_KERNEL_ERROR_UNKNOWN_THREAD_ID`.
Or is it a copy-paste from our `sceKernelChangeThreadPriority` or `sceKernelGetThreadInfo` implementation?

Comment (bookmist) [reply vita3k/modules/SceLibKernel/SceLibKernel.cpp:1105]: It's a copy-paste.

Comment (scribam) [reply vita3k/modules/SceLibKernel/SceLibKernel.cpp:1105]: Ok, it should be fine as it is. It can be improved afterwards.

**Comment by bookmist** [comment]
> But sceKernelGetLwMutexInfoById and sceKernelGetLwMutexInfo keep missing
They will be in next PR bookmist:sceKernelGetLwMutexInfo Previous big PR was not finished, so now i'm going to make small steps.
I hope this PR will be merged someday.

For "sceKernelCheckCallback" Any game, using it? (I need sample to check)

**Comment by Zangetsu38** [comment]
PCSB01011 VALKYRIE DRIVE -BHIKKHUNI-
 using it on boot

File: vita3k/modules/SceLibKernel/SceLibKernel.cpp (hunk 1 above)
Comment (frangarcj) [reply vita3k/modules/SceLibKernel/SceLibKernel.cpp:1105]: Null or 0 means current thread

**Pull request merged**